python/sbp/msg.py

```diff
@@ -21,44 +20,84 @@
 from sbp.constants import SBP_PREAMBLE as _SBP_PREAMBLE
 from sbp.constants import crc16_tab
 
-import numpy as np
 
 from pkgutil import iter_modules
 
 import sys
 
-parse_jit_name = "parse_jit_py{}".format(str(sys.version_info[0]) + str(sys.version_info[1]))
+SENDER_ID = _SENDER_ID
+SBP_PREAMBLE = _SBP_PREAMBLE
+
+NOJIT = False
+NONUMPY = False
+
+try:
+  import importlib
+  import numpy as np
+except ImportError:
+  NONUMPY = True
 
-if parse_jit_name in (name for loader, name, ispkg in iter_modules()):
+parse_jit_crc16 = None
+parse_jit = None
+
+
+def try_import_jit():
+  parse_jit_name = "parse_jit_py{}".format(str(sys.version_info[0]) + str(sys.version_info[1]))
+  if parse_jit_name in (name for loader, name, ispkg in iter_modules()):
     # found in sys.path
     parse_jit = importlib.import_module(parse_jit_name)
-elif parse_jit_name in (name for loader, name, ispkg in iter_modules(['sbp/jit'])):
+  elif parse_jit_name in (name for loader, name, ispkg in iter_modules(['sbp/jit'])):
     # found in sbp.jit
     parse_jit = importlib.import_module('sbp.jit.' + parse_jit_name)
-else:
+  else:
     # not found -> compile
-    from sbp.jit import parse
-    parse.compile()
-    parse_jit = importlib.import_module('sbp.jit.' + parse_jit_name)
-
-
-np_crc16_tab = np.array(crc16_tab, dtype=np.uint16)
-
-SENDER_ID = _SENDER_ID
-SBP_PREAMBLE = _SBP_PREAMBLE
-
-crc_buffer = np.zeros(512, dtype=np.uint8)
+    try:
+      from sbp.jit import parse
+      parse.compile()
+      parse_jit = importlib.import_module('sbp.jit.' + parse_jit_name)
+    except ImportError:
```

Comment:
Maybe if user gets here we can add some helpful hints to a user.

1. Indicate which libraries / files were missing so that if someone is messing with pyinstaller they might realize that they forget to include the right stuff and won't need jit if they just get the right files. 
1. Refer a user to the last version of libsbp that didn't require jit compilation / numba.  In that way people on odd systems who can't get these dependencies easily can just fallback to a simpler (but slower) libsbp.

Reply:
Item #2 should not be necessary with this change, we should automatically just fallback to parsing without Numba/Numpy.

Reply:
FYI, for item 1 I haven't been able to figure out how to get PyInstaller to work with the Numba/JIT stuff yet (I've been able to get PyInstaller to package the library, but once the app starts we can't find it), as you saw recent versions of the library broke things like the Windows build, so this "graceful fallback" is necessary for these cases for now.

Reply:
I think instead of a warning, setting a property within the library might be appropriate?  So if someone really needs the speed from the JIT code, they can assert on that property.